HunterPie.Integrations/Datasources/Common/Entity/Player/CommonPlayer.cs

```diff
@@ -110,6 +112,13 @@ public abstract class CommonPlayer : Scannable, IPlayer, IEventDispatcher, IDisp
         remove => _onLevelChange.Unhook(value);
     }
 
+    protected readonly SmartEvent<EventArgs> _onCombatStatusChange = new();
```

Comment:
You need to dispose `SmartEvent<T>`, it'll create a memory leak otherwise.

https://github.com/HunterPie/HunterPie/blob/e79484bd7748d33b118219d9e076ca6e63daaf08/HunterPie.Integrations/Datasources/Common/Entity/Player/CommonPlayer.cs#L177-L183